src/main/java/duke/utility/Storage.java

```diff
@@ -0,0 +1,126 @@
+                    + "\nTask log will be saved there.");
+        }
+    }
+
+    /**
+     * Parses the locally saved task log from the file path provided and generates a list of tasks to import.
+     * @return a list of tasks generated from parsing the locally saved task log.
+     * @throws FileNotFoundException if the stipulated file can neither be read from (e.g. does not exist) nor
+     * be created.
+     */
+    public List<Task> loadPreviousTasks() throws FileNotFoundException {
+        List<Task> prevTasks = new ArrayList<Task>();
+        Scanner sc = new Scanner(new File(this.filePath));
+        while (sc.hasNextLine()) {
+            String task = sc.nextLine();
+            String[] tokens = task.split(";");
+            boolean isCompleted = tokens[1].equals("T");
+            String taskName = tokens[2];
+            // format: duke.task type | isCompleted | event name | date/time
+            switch (tokens[0]) {
+            case "T":
+                prevTasks.add(ToDo.createTask(taskName, isCompleted));
+                break;
+            case "E":
+                prevTasks.add(Event.createTask(taskName, isCompleted, LocalDateTime.parse(tokens[3].trim(),
+                        DateTimeFormatter.ISO_LOCAL_DATE_TIME)));
+                break;
+            default: // "D"
+                prevTasks.add(Deadline.createTask(taskName, isCompleted, LocalDateTime.parse(tokens[3].trim(),
+                        DateTimeFormatter.ISO_LOCAL_DATE_TIME)));
+                break;
+            };
+        }
+        return prevTasks;
+    }
+
+    void append(String e, String f, String taskName, String dateTime) throws IOException {
+        this.append(e, f, taskName + ";" + dateTime);
+    }
+
+    void append(String type, String isCompleted, String detail) throws IOException {
+        FileWriter fw = new FileWriter(this.filePath, true); // append flag true -> append, not overwrite
+        fw.write(type + ";" + isCompleted + ";" + detail + "\n");
+        fw.close();
+    }
+
+    void changeTaskLogToCompleted(int lineNum) throws IOException {
+        int currentLine = 0;
+        Scanner sc = new Scanner(new File(this.filePath));
+        StringBuilder sb = new StringBuilder();
+        while (sc.hasNextLine()) { // read the entire file except the line to change, which is ignored
+            String entry = sc.nextLine();
+            if (currentLine != lineNum) { // line to be modified
+                sb.append(entry);
+            } else { // generate new entry
+                sb.append(entry.replaceAll(";F;", ";T;"));
+            }
+            sb.append("\n");
+            currentLine++;
+        }
+        FileWriter fw = new FileWriter(this.filePath, false); // append flag false -> overwrite file
+        fw.write(sb.toString());
+        fw.close();
+        sc.close();
+    }
+
+    void deleteTaskLogEntry(int lineIdx) throws IOException {
```

Comment:
Perhaps the variable lineIdx could be explained in some Javadocs or maybe the variable name could be more descriptive? It may not be obvious what Idx means to readers.